addons/sale/wizard/payment_link_wizard.py

```diff
@@ -32,9 +32,18 @@ def _prepare_query_params(self, *args):
         if self.res_model != 'sale.order':
             return res
 
+        sale_order = self.env['sale.order'].browse(self.res_id)
         # The other order-related values are read directly from the sales order in the controller.
         return {
-            'amount': self.amount,
-            'access_token': self._prepare_access_token(),
-            'sale_order_id': self.res_id,
+            'link_amount': self.amount,
+            'access_token': sale_order.access_token or sale_order._portal_ensure_token(),
```

Comment:
Why did not you override `_prepare_access_token`   instead?

Reply:
I don't see added value of overriding _prepare_access_token to just return _portal_ensure_token if it's Sale order, and call super in other case. But the sale_order.access_token can be removed indeed.